### ISSUE TITLE
Question: Must condition inside augment module failing?

### ISSUE THREAD

**Issue by Roky7250** [sysrepo/sysrepo#2076]
Hi,
I am using newest sysrepo, libyang, libnetconf2 and netopeer2 from master branches.
sysrepo version: v1.4.70
netopeer2 version: 1.1.39

I am using ietf-interfaces and ieee802-dot1q-bridge yang modules. I created a my-test.yang module which expands container bridge-port inside ieee802-dot1q-bridge yang module as augment (augment "/if:interfaces/if:interface/dot1q:bridge-port").

So container bridge-port is augment of list interface (augment "/if:interfaces/if:interface") and container my-test-cont from my module my-test.yang i augment of container bridge-port (augment "/if:interfaces/if:interface/dot1q:bridge-port").

Main part of module looks like this:
```
  augment "/if:interfaces/if:interface/dot1q:bridge-port" {
    description
      "test augment.";
    container my-test-cont {
      description
        "test container.";
      container my-list {
        must "(count(./my-test-entry) > 0)" {
          error-message "my-test-entry list empty.";
        }
		config true;
        description
          "test container.";
        uses my-test-entries;
      }
    }
  }
```

I use this must condition to check if `my-test-entry` list inside container my-list is empty.

Problem is next. When  startup datastore is filled up, sysrepo-plugind crashes every time on this must condition  `must "(count(./my-test-entry) > 0)"`, with error message:

`[ERR]: Must condition "(count(./my-test-entry) > 0)" not satisfied. (path: /ietf-interfaces:interfaces/interface[name='lo']/ieee802-dot1q-bridge:bridge-port/my-test:my-test-cont/my-list)`

So my question is next, why is this must condition failing when there is no container `my-test-cont` inside container `bridge-port` for interface "lo"?

Inside code I have condition that checks if some condition is fulfilled then create container `my-test-cont` and other containers and list inside. I would expect that this must condition fails if there is a container `my-test-cont` and `my-test` but must condition for list `my-test-entry` is not fulfilled. 

When I changed must statement to:
```
        must "(count(./my-test-entry) >= 0)" {
          error-message "my-test-entry list empty.";
        }
```
everything went fine, startup datastore was filled up.

Best regards

**Comment by michalvasko** [comment]
Hi Roky,
so you have, yet again, discovered the ambiguity of non-presence containers and the fact that they are specified as having no semantic meaning but at the same time often having semantic meaning (for example, when they have a `must` condition).

In short, *libyang* (and hence *sysrepo* as well) treats NP containers similar to default values meaning they exist as long as they parent exists (so they are created automatically). So, all I can then say is that the design of your YANG is flawed in this respect and you should fix it, somehow. However, I am not exactly how, but I suppose adding `presence` for the container is one way (when you essentially give meaning to the container because it has one in your use-case).

Finally, note that `must` condition `must "(count(./my-test-entry) >= 0)"` has no meaning and is always true. Secondly, there is a special statement `min-elements` that you can use for restricting number of list instances, no need to use the generic `must`.

Regards,
Michal

**Comment by Roky7250** [comment]
Hi, I created this **my-test.yang** file just to explain to issue. The original yang file has same structure and its practical purpose.

So,  _libyang_ (and hence _sysrepo_ as well) are creating NP container automatically if their parent exists.
In order to stop sysrepo to do that, i need to put `presence my-test;` inside `my-test` container, to disable sysrepo creating that container automatically. Using that I am creating P container and telling sysrepo that container's existence in the data tree carries some meaning. That container is not there like NP container who has no meaning of its own and exists only to contain child nodes.

I added  `presence my-test;` inside container `my-test` and everything worked fine, sysrepo-plugind boot fine and startup configuration was created. I knew that sysrepo was creating leafs with default values automatically, but I did not know that it also creates NP container automatically. I looked me code and DBG messages and I saw that I was not explicitly creating containers `my-test-cont` and `my-test` using **lyd_new** functions, so I was confused why is condition failing when they do not exist.

Thanks for help
Regards
